src/notebooklm/auth.py

```
@@ -642,22 +644,55 @@ def load_httpx_cookies(path: Path | None = None) -> "httpx.Cookies":
     Raises:
-        httpx.HTTPError: If request fails
-        ValueError: If tokens cannot be extracted from response
+        RuntimeError: If the refresh command is missing, times out, or exits
+            non-zero.
     """
+    cmd = os.environ.get(NOTEBOOKLM_REFRESH_CMD_ENV)
+    if not cmd:
+        raise RuntimeError(f"{NOTEBOOKLM_REFRESH_CMD_ENV} is not set; cannot refresh cookies.")
+    os.environ[_REFRESH_ATTEMPTED_ENV] = "1"
+    try:
+        result = subprocess.run(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)